src/Microsoft.Diagnostics.Runtime/src/Common/HeapBase.cs

```diff
@@ -235,24 +235,36 @@ protected internal override IEnumerable<ClrObject> EnumerateObjectReferences(ulo
             else
                 Debug.Assert(type == GetObjectType(obj));
 
-            if (type == null || !type.ContainsPointers)
+            if (type == null || (!type.ContainsPointers && !type.IsCollectible))
                 return s_emptyObjectSet;
 
-            GCDesc gcdesc = type.GCDesc;
-            if (gcdesc == null)
-                return s_emptyObjectSet;
+            List<ClrObject> result = null;
 
-            ulong size = type.GetSize(obj);
-            if (carefully)
+            if (type.ContainsPointers)
             {
-                ClrSegment seg = GetSegmentByAddress(obj);
-                if (seg == null || obj + size > seg.End || (!seg.IsLarge && size > 85000))
+                GCDesc gcdesc = type.GCDesc;
+                if (gcdesc == null)
                     return s_emptyObjectSet;
+
+                ulong size = type.GetSize(obj);
+                if (carefully)
+                {
+                    ClrSegment seg = GetSegmentByAddress(obj);
+                    if (seg == null || obj + size > seg.End || (!seg.IsLarge && size > 85000))
+                        return s_emptyObjectSet;
+                }
+
+                result = new List<ClrObject>();
+                MemoryReader reader = GetMemoryReaderForAddress(obj);
+                gcdesc.WalkObject(obj, size, ptr => ReadPointer(reader, ptr), (reference, offset) => result.Add(new ClrObject(reference, GetObjectType(reference))));
+            }
+
+            if (type.IsCollectible)
+            {
+                result ??= new List<ClrObject>(1);
```

Comment:
Can you please change all instances of this pattern to `result = result ?? ...`?  While this is a valid C# feature, it apparently breaks Arcade (the build system used to produce signed builds).  I haven't tracked down that problem yet, but if left as-is I'll have to go fix these on a second checkin.

Reply:
Can you enable CI on PRs so contributors can check for any build or test failures?

Reply:
Reverting, because Arcade has been updated in #316 to use .NET Core SDK 3.0 with C# 8.0 available, so I believe this should build successfully.

Reply:
I will look into it after this current sprint of work.  I would like to have it on as well, but I'm not sure the process there.